src/main/java/net/glowstone/entity/GlowEntity.java

```diff
@@ -605,6 +615,40 @@ public void pulse() {
                 }
             }
         }
+        GlowBlock block = world.getBlockAt(location);
+        if (block.getType().equals(Material.PORTAL)
+            && BlockPortal.getBoundingBox(block).intersects(boundingBox)) {
+            if (!isInPortal) {
```

Comment:
Could you document this code block overall? It is a bit hard to follow.

Reply:
Applied